src/handlers/choosingArticle.js

```diff
@@ -133,7 +137,12 @@ export default async function choosingArticle(params) {
     ];
 
     if (articleReplies.length !== 0) {
-      data.foundReplyIds = articleReplies.map(({ reply }) => reply.id);
+      data.foundReplyIds = articleReplies.map(({ reply }) => {
+        // Track which Reply is searched.
+        ga(userId, { ec: 'Reply', ea: 'Search', el: reply.id });
```

Comment:
`.map` calls are not supposed to contain side effects (such as calling other APIs).

Let's do impression tracking in a dedicated for-each loop. We can put this along side with the `Article/Selected` event above, so that similar logics are located in the same place.

Also, since we are sending one Reply/Search event for each found replies, let's make it [non-interactive](https://support.google.com/analytics/answer/1033068#NonInteractionEvents) so that we can get better bounce rate calculations. We can achieve this by using [`ni`](https://github.com/peaksandpies/universal-analytics/blob/master/AcceptableParams.md#non-interaction-hit) param.